app/api/business/brief_overview_business.py

```diff
@@ -1,8 +1,69 @@
 from flask_login import current_user
+
 from app.api.business.agreement_business import use_old_work_order_creator
+from app.api.helpers import abort
+from app.api.services import (audit_service, audit_types,
+                              brief_responses_service, briefs)
+from app.emails import send_opportunity_closed_early_email
+from app.tasks import publish_tasks
+from app.tasks.s3 import create_responses_zip
 from app.validation import get_sections as get_validation_sections
 
 
+def can_close_opportunity_early(brief):
+    seller_selector = brief.data.get('sellerSelector', '')
```

Comment:
The `/close` POST endpoint + `can_close_opportunity_early()` + `close_opportunity_early()` is currently lacking validation that the requesting user has permissions to close this specific brief. It checks that the user is a buyer and that the buyer user has the publish opportunities permission, but there is no check for whether the brief is owner by the buyer user or the buyer user's team. This is done via `briefs.has_permission_to_brief(user_id, brief_id)` typically.

I'm not sure if this is the best spot for that check, as `can_close_opportunity_early()` may be called in a context where there is no current user. So `can_close_opportunity_early()` may need to accept the current user as an argument, and perhaps the `has_permission_to_brief()` check should be done in the POST endpoint (where it should respond with a `forbidden()` on failure).

Reply:
Thanks for picking this up!  I've added the additional permissions check in `close_opportunity_early` and raise an error which is caught by the view.